kasa/smart/smartdevice.py

```diff
@@ -158,17 +167,27 @@ async def _initialize_modules(self):
         """Initialize modules based on component negotiation response."""
         from .smartmodule import SmartModule
 
+        if self._parent and self._parent.device_type == DeviceType.WallSwitch:
+            modules = self._parent.modules
+            skip_parent_only_modules = True
+        else:
+            modules = self.modules
+            skip_parent_only_modules = False
+
         for mod in SmartModule.REGISTERED_MODULES.values():
             _LOGGER.debug("%s requires %s", mod, mod.REQUIRED_COMPONENT)
+
+            if skip_parent_only_modules and mod in PARENT_ONLY_MODULES:
+                continue
```

Comment:
This will break "Firmware" for P300, for both the main device and for the children, too:
```
                Auto update enabled: got exception (You need to call update() prior accessing module data for 'firmware')
                Update available: got exception (You need to call update() prior accessing module data for 'firmware')

```

Reply:
The P300 isn’t a `WallSwitch` so not sure why this is happening. Will try to look unless you have an idea.

Reply:
Looks like this is caused by some changes in the error handling. In the current master, if a call errors out, the error will be stored in the internal state -> the check for update passes.

Reply:
Not related to this PR but fixed with https://github.com/python-kasa/python-kasa/pull/854